plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -161,18 +167,27 @@ class CodeWhispererService(private val cs: CoroutineScope) : Disposable {
         val isInjectedFile = runReadAction { psiFile.isInjectedText() }
         if (isInjectedFile) return
 
+        val currentJobId = jobId++
         val requestContext = try {
-            getRequestContext(triggerTypeInfo, editor, project, psiFile, latencyContext)
+            getRequestContext(triggerTypeInfo, editor, project, psiFile)
         } catch (e: Exception) {
             LOG.debug { e.message.toString() }
             CodeWhispererTelemetryService.getInstance().sendFailedServiceInvocationEvent(project, e::class.simpleName)
             return
         }
+        val caretContext = requestContext.fileContextInfo.caretContext
+        ongoingRequestsContext.forEach { (k, v) ->
```

Comment:
for example, this is guaranteed to iterate in-order from first request inserted in the map to last

Reply:
yea, the purpose is not just to identify which the last request is -- but more to separate states of different triggers.

Later on when we send UserTriggerDecisions we will need to send one for each jobId(trigger), having a jobId here and logging it keeps track of what happened E2E for this specific trigger.